File: praticas_r/04-chatgpt.R
# usethis::edit_r_environ("project")

library(httr)

u_base <- "https://api.openai.com/v1/"
endpoint <- "chat/completions"
u_completions <- paste0(u_base, endpoint)

api_key <- Sys.getenv("OPENAI_API_KEY")

headers <- add_headers(
  Authorization = paste0("Bearer ", api_key)
)

# data = '{\n     "model": "gpt-4o-mini",\n     "messages": [{"role": "user", "content": "Say this is a test!"}],\n     "temperature": 0.7\n   }'

data <- list(
  model = "gpt-4o-mini",
  messages = list(
    list(
      role = "user",
      "content" = "Diga que isso é um teste!!"
    )
  ),
  temperature = 0.7
)

res <- POST(
  u_completions, 
  headers, 
  #content_type_json(),
  body = data,
  encode = "json"
)

res |> 
  content() |> 
  purrr::pluck("choices", 1, "message", "content")

File: praticas_py/04-chatgpt.py
# %% carregando bibliotecas e variáveis de ambiente
import requests
from dotenv import load_dotenv
import os
load_dotenv('./.env')

openai_key = os.getenv('OPENAI_API_KEY')

# %% dados a serem enviados
endpoint = 'https://api.openai.com/v1/chat/completions'

instrucao_inicial = 'Você é um poeta que escreve no estilo de Olavo Bilac.'

prompt = 'O rato roeu a roupa do rei de roma?'

messages = [
  {
    'role': 'system',
    'content': instrucao_inicial
  },
  {
    'role':
    'user', 'content': prompt
  }
]

# %% fazendo a chamada

headers = {
  'Content-Type': 'application/json',
  'Authorization': 'Bearer ' + openai_key
}

payload = {
  'model': 'gpt-4-turbo',
  'messages': messages,
  'temperature': 0.5
}

r = requests.post(endpoint, headers=headers, json=payload)

# %% verificando o resultado
print(r.json()['choices'][0]['message']['content'])